src/main/java/pedido/Pedido.java

```diff
@@ -1,51 +1,113 @@
+    public Cliente getCliente() {
         return this.cliente;
     }
 
-    public double calcularTotal(Cardapio cardapio){
-        double total= 0;
-        //TODO
+    public double calcularTotal(Cardapio cardapio) {
+        Double total = (double) 0;
+        for (ItemPedido itemPedido : this.itens) {
+            Double totalItem = (double) 0;
+            for (Map.Entry<Ingrediente, Double> precoIngrediente : cardapio.getPrecos().entrySet()) {
+                if (itemPedido.getShake().getBase().compareTo(precoIngrediente.getKey()) == 0) {
+                    totalItem += precoIngrediente.getValue();
+                }
+            }
+            totalItem = totalItem * itemPedido.getShake().getTipoTamanho().multiplicador;
+            for (Map.Entry<Ingrediente, Double> precoIngrediente : cardapio.getPrecos().entrySet()) {
+                for (Adicional adicional : itemPedido.getShake().getAdicionais()) {
+                    if (adicional.equals(precoIngrediente.getKey())) {
+                        totalItem += precoIngrediente.getValue();
+                    }
+                }
+            }
+            totalItem = totalItem * itemPedido.getQuantidade();
+            total += totalItem;
+        }
         return total;
     }
 
-    public void adicionarItemPedido(ItemPedido itemPedidoAdicionado){
-        //TODO
+    public void adicionarItemPedido(ItemPedido itemPedidoAdicionado) {
+        sortAdicionais(itemPedidoAdicionado);
+        boolean added = false;
+        for (ItemPedido item : itens) {
+            if (!added) {
+                if (item.getShake().equals(itemPedidoAdicionado.getShake())) {
+                    item.setQuantidade(item.getQuantidade() + itemPedidoAdicionado.getQuantidade());
+                    added = true;
+                }
+            }
+        }
+
+        if (!added) {
+            itens.add(itemPedidoAdicionado);
+        }
     }
 
     public boolean removeItemPedido(ItemPedido itemPedidoRemovido) {
-        //substitua o true por uma condição
-        if (true) {
-            //TODO
-        } else {
-            throw new IllegalArgumentException("Item nao existe no pedido.");
+        sortAdicionais(itemPedidoRemovido);
+        ItemPedido itemToRemove = null;
+        boolean removed = false;
+        for (ItemPedido item : itens) {
+            if (!removed) {
+                if (item.getShake().equals(itemPedidoRemovido.getShake())) {
+                    if (item.getQuantidade() == 1) {
+                        itemToRemove = item;
+                        removed = true;
+                    } else {
+                        item.setQuantidade(item.getQuantidade() - 1);
+                        removed = true;
+                    }
+                }
+            }
         }
+        if (itemToRemove != null)
+            itens.remove(itemToRemove);
+        if (!removed)
+            throw new IllegalArgumentException("Item nao existe no pedido.");
         return false;
```

Comment:
Bem pensado! Digo o mesmo do método de cima, caso queira retirar o uso do for, da para fazer essa lógica bem simples utilizando indexOf.